src/layer/AbstractLayer.ts

```diff
@@ -30,7 +30,12 @@ export class AbstractLayer {
     this.legendUrl = legendUrl;
   }
 
-  public async getMap(params: GetMapParams, api: ApiType, reqConfig?: RequestConfiguration): Promise<Blob> {
+  @timeoutWrapper(2)
```

Comment:
As we have discussed recently, I think it would be better to avoid using the decorators altogether, not make it more explicit how the `reqConfig` parameter changes. 

The suggestion was:
```typescript
public async getMap(params: GetMapParams, api: ApiType, reqConfigWithTimeout?: RequestConfiguration): Promise<Blob> {
  return await ensureTimeout(reqConfigWithTimeout, async (reqConfig) => {
    ...
  })
}
```

Then `ensureTimeout` can look similar to this:
```javascript
async function ensureTimeout(originalReqConfig, innerFunc) {
  if (!originalReqConfig) {
    return await innerFunc(originalReqConfig);
  }

  // setTimeout, clearTimeout,...

  return newReqConfig;
}
```

This would make implementation much easier (imho) and explicit about what is going on. What do you think?